packages/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -0,0 +1,527 @@
+      v-if="!hidePopover"
+      #content>
+      <KSegmentedControl
+        v-if="hasRelativeTimeframes && hasCalendar"
+        v-model="tabName"
+        :options="[
+          { label: 'Relative', value: 'relative' },
+          { label: 'Custom', value: 'custom' }
+        ]"
+        class="w-100 mb-5"
+        data-testid="analytics-time-toggle"
+      />
+      <!-- Human timestamp readout -->
+      <p v-if="!showCalendar">{{ fullRangeDisplay }}</p>
+      <DatePicker
+        v-if="hasCalendar && showCalendar"
+        v-model="selectedCalendarRange"
+        :is-range="range"
+        :max-date="maxDate"
+        :min-date="minDate"
+        :mode="mode"
+        :model-config="modelConfig"
+        :select-attribute="calendarSelectAttributes"
+        is-expanded
+      />
+      <div
+        v-else-if="hasRelativeTimeframes && hasTimePeriods"
+        class="d-flex flex-column"
+      >
+        <div
+          v-for="(item, index) in timePeriods"
+          :key="`section-${index}`"
+          class="timeframe-section d-flex flex-column"
+        >
+          <h4>{{ item.section }}</h4>
+          <div class="timeframe-buttons d-flex flex-row justify-content-start">
+            <KButton
+              v-for="(timeFrame, index) in item.values"
+              :key="`time-${index}`"
+              :is-rounded="false"
+              :class="{'selected-option': timeFrame.key === selectedTimeframe.key}"
+              :data-testid="'select-timeframe-' + timeFrame.timeframeLength()"
+              appearance="outline"
+              size="medium"
+              @click="changeRelativeTimeframe(timeFrame)"
+            >
+              {{ ucWord(timeFrame.timeframeText) }}
+            </KButton>
+          </div>
+        </div>
+      </div>
+    </template>
+    <template
+      v-if="!hidePopover"
+      #footer>
+      <div class="d-flex justify-content-end">
+        <KButton
+          :is-rounded="false"
+          size="medium"
+          appearance="btn-link"
+          @click="clearSelection()"
+        >
+          Clear
+        </KButton>
+        <KButton
+          :disabled="submitDisabled"
+          :is-rounded="false"
+          size="medium"
+          appearance="btn-link"
+          @click="submitTimeFrame()"
+        >
+          Apply
+        </KButton>
+      </div>
+    </template>
+  </KPop>
+</template>
+
+<script>
+import KButton from '@kongponents/kbutton/KButton.vue'
+import KPop from '@kongponents/kpop/KPop.vue'
+import KSegmentedControl from '@kongponents/ksegmentedcontrol/KSegmentedControl.vue'
+import { format } from 'date-fns'
+
+export default {
+  name: 'KTimePicker',
+  components: {
+    KButton,
+    KPop,
+    KSegmentedControl,
+    DatePicker: () => import('v-calendar/lib/components/date-picker.umd')
+  },
+  props: {
+    placeholder: {
+      type: String,
+      required: false,
+      default: 'Select a time range'
+    },
+    mode: {
+      type: String,
+      required: false,
```

Comment:
I'm not sure I understand why `mode` isn't a required prop? Is it ever valid if it's an empty string?

Reply:
I guess passing in `relative` for `mode` when we want a timeframe-only instance should do the trick (we just have to make sure we never pass that to `v-calendar` or it will self-implode).  See my latest commit plz:

https://github.com/Kong/kongponents/pull/768/commits/03a802aaf22fcf522ce5b5d8ef52cff96d2fd168?w=1#diff-fa2c62a1ce67b2096a11c57505e380b633881e8a3ff68009f96fba3f8676074cR122